### PR TITLE
Fix errors in service account template.

### PR DESCRIPTION
**Pull request by v-shenoy**
Signed-off-by: Vighnesh Shenoy <vshenoy@microsoft.com>

_Provide a description of what has been changed_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*
- [x] README is updated with new configuration values *(if applicable)*

Relates to https://github.com/kedacore/keda/issues/2487
Relates to https://github.com/kedacore/charts/pull/263

### CODE REVIEW

**Comment by v-shenoy** [comment]
@tomkerkhove Kinda new to helm charts, so I didn't realize I could install from local directories. So I didn't test my changes until I pushed them to the PR. But I realized I made some errors, please merge otherwise the helm changes will throw errors.

File: keda/templates/01-serviceaccount.yaml [hunk 1]
```diff
@@ -11,9 +11,11 @@ metadata:
     {{- if .Values.podIdentity.azureWorkload.enabled }}
     azure.workload.identity/client-id: {{ .Values.podIdentity.azureWorkload.clientId }}
     azure.workload.identity/tenant-id: {{ .Values.podIdentity.azureWorkload.tenantId }}
-    azure.workload.identity/service-account-token-expiration: {{ .Values.podIdentity.azureWorkload.tokenExpiration }}
+    azure.workload.identity/service-account-token-expiration: "{{ .Values.podIdentity.azureWorkload.tokenExpiration }}"
```
Comment (tomkerkhove) [inline keda/templates/01-serviceaccount.yaml:14]: Helm also allows you to do this:
```suggestion
    azure.workload.identity/service-account-token-expiration: {{ .Values.podIdentity.azureWorkload.tokenExpiration | quote }}
```

**Review by tomkerkhove — APPROVED** [review]

**Comment by tomkerkhove** [comment]
No problem at all! Normally the CI should have caught that but we don't test all configuration scenarios, unfortunately.

**Comment by tomkerkhove** [comment]
Let's not merge this PR just yet.

**Comment by tomkerkhove** [comment]
The convention is `{{ .Values.podIdentity.azureWorkload.tokenExpiration | quote }}` over  `{{ quote 
 .Values.podIdentity.azureWorkload.tokenExpiration }}` but that's OK.

**Pull request merged**